### PR TITLE
Added TypeScript Webpack React Template

### PR DESCRIPTION
**Pull request by JFenlonWork**

### CODE REVIEW

File: Templates/TypeScriptWebpackReactVisualizationTemplate/src/visualization01/src/types/actions.d.ts
Comment (JFenlonWork) [inline Templates/TypeScriptWebpackReactVisualizationTemplate/src/visualization01/src/types/actions.d.ts:1]: Cant comment inside the file as it is auto-generated.
Declare global may work differently than I can figure out, but the purpose of it is to allow the Vis.Actions type to be usable inside the visualizations source code without requiring an import.
The file is only picked up by the surrounding tsconfig which should mean it doesn't spread to other visualization folders.

**Review by AWoodhouseAtCACI — APPROVED** [review]

**Review by Gilesey — APPROVED** [review]
I'm happy if Andrew is happy

File: Templates/TypeScriptWebpackReactVisualizationTemplate/helpers/config.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  //Send an event to any useStyle hook listeners
+  document.dispatchEvent(new CustomEvent(updateVisualizationStyleEventKey));
+};
+
+/**
+ * Clone and deep freeze the visualization style into frozenVisualizationStyle
+ */
+export const updateFrozenVisualizationStyle = () => {
+  const styleClone = structuredClone(getVisualizationConfig().style);
+  deepFreeze(styleClone);
```
[the remark this answers is not in the document]
Comment (JFenlonWork) [reply Templates/TypeScriptWebpackReactVisualizationTemplate/helpers/config.ts:132]: deepFreeze now returns undefined if the provided object is null/undefined, the types of all of the frozen values are marked as potentially undefined so the users are informed (although these are mostly used internally)

File: Templates/TypeScriptWebpackReactVisualizationTemplate/typescript-transformer/transform-datashape.js [hunk 2]
```diff
@@ -0,0 +1,274 @@
+const glob = require("glob");
```
[the remark this answers is not in the document]
Comment (JFenlonWork) [reply Templates/TypeScriptWebpackReactVisualizationTemplate/typescript-transformer/transform-datashape.js:1]: This is a valid suggestion, however it seems that those implementations requre a GraphQL schema which was originally why I rolled my own

**Pull request merged**